### PR TITLE
Fix of Hybrid Router after asset system merge

### PR DESCRIPTION
**Pull request by Chralt98**
### What does it do?

After there happened some commits to the main branch, I needed to adjust the Hybrid Router implementation again to have a working version.

### CODE REVIEW

File: primitives/src/market.rs [hunk 1]
```diff
@@ -182,10 +179,7 @@ impl<AI, BA, BN, M, A> Market<AI, BA, BN, M, A> {
     ) -> Option<MarketAssetClass<MI>> {
         match outcome_report {
             OutcomeReport::Categorical(idx) => match self.scoring_rule {
-                ScoringRule::Orderbook => {
-                    Some(MarketAssetClass::<MI>::CategoricalOutcome(market_id, *idx))
-                }
-                ScoringRule::Lmsr => {
+                ScoringRule::AmmCdaHybrid => {
```
Comment (sea212) [inline primitives/src/market.rs:182]: It will be impossible to use orderbook / lmsr only?

File: zrml/hybrid-router/Cargo.toml [hunk 2]
```diff
@@ -51,6 +54,8 @@ mock = [
     "sp-io/default",
     "xcm/default",
     "zeitgeist-primitives/mock",
+    "zrml-asset-router/default",
+    "zrml-asset-router/std",
```
Comment (sea212) [inline zrml/hybrid-router/Cargo.toml:58]:
```suggestion
```

File: zrml/hybrid-router/src/lib.rs [hunk 3]
```diff
@@ -67,7 +67,7 @@ mod pallet {
         },
         orderbook::{Order, OrderId},
         traits::{HybridRouterAmmApi, HybridRouterOrderbookApi},
-        types::{Asset, Market, MarketType, ScalarPosition},
+        types::{Asset, BaseAsset, BaseAssetClass, Market, MarketType, ScalarPosition},
```
Comment (sea212) [inline zrml/hybrid-router/src/lib.rs:70]:
```suggestion
        types::{Asset, BaseAsset, Market, MarketType, ScalarPosition},
```

File: zrml/hybrid-router/src/lib.rs [hunk 4]
```diff
@@ -455,7 +455,7 @@ mod pallet {
             mut remaining: BalanceOf<T>,
             who: &AccountIdOf<T>,
             market_id: MarketIdOf<T>,
-            base_asset: AssetOf<T>,
+            base_asset: BaseAssetClass,
```
Comment (sea212) [inline zrml/hybrid-router/src/lib.rs:458]:
```suggestion
            base_asset: BaseAsset,
```

File: zrml/hybrid-router/src/tests/mod.rs [hunk 5]
```diff
@@ -27,8 +27,8 @@ use zeitgeist_primitives::{
     constants::{base_multiples::*, BASE, CENT},
     orderbook::Order,
     types::{
-        AccountIdTest, Asset, Deadlines, MarketCreation, MarketPeriod, MarketStatus, MarketType,
-        MultiHash, ScoringRule,
+        AccountIdTest, Asset, Assets, Deadlines, MarketCreation, MarketPeriod, MarketStatus,
```
Comment (sea212) [inline zrml/hybrid-router/src/tests/mod.rs:30]:
```suggestion
        AccountIdTest, Assets, Deadlines, MarketCreation, MarketPeriod, MarketStatus,
```

File: primitives/src/market.rs (hunk 1 above)
Comment (Chralt98) [reply primitives/src/market.rs:182]: Both just use the scoring rule AmmCdaHybrid.

File: zrml/hybrid-router/Cargo.toml (hunk 2 above)
Comment (Chralt98) [reply zrml/hybrid-router/Cargo.toml]: 20dd0f08d202a93969470fb76fcefdeceaecabcd

File: zrml/hybrid-router/src/lib.rs (hunk 3 above)
Comment (Chralt98) [reply zrml/hybrid-router/src/lib.rs]: 20dd0f08d202a93969470fb76fcefdeceaecabcd

File: zrml/hybrid-router/src/tests/mod.rs (hunk 5 above)
Comment (Chralt98) [reply zrml/hybrid-router/src/tests/mod.rs]: 20dd0f08d202a93969470fb76fcefdeceaecabcd

**Review by sea212 — APPROVED** [review]

**Pull request merged**